src/Compilers/CSharp/Portable/Symbols/Source/SourceDestructorSymbol.cs

```diff
@@ -33,7 +36,7 @@ internal sealed class SourceDestructorSymbol : SourceMethodSymbol
                 }
             }
 
-            if (!modifierErrors && bodySyntaxReferenceOpt == null && !IsExtern)
+            if (!modifierErrors && bodySyntaxReferenceOpt == null && !IsExtern && !IsExpressionBodied)
```

Comment:
Why do we have to adjust this condition? Isn't checking bodySyntaxReferenceOpt sufficient?